### PR TITLE
Improve message when a S3 bucket is not found or the permission to the bucket is denied

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
Signed-off-by: Hanwen <hanwenli@amazon.com>

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/validators/s3_validators.py [hunk 1]
```diff
@@ -86,7 +86,17 @@ def _validate(self, url):
                 bucket = get_bucket_name_from_s3_url(url)
                 AWSApi.instance().s3.head_bucket(bucket_name=bucket)
             except AWSClientError as e:
-                self._add_failure(str(e), FailureLevel.ERROR)
+                if e.error_code == "404":
+                    self._add_failure(f"Bucket {bucket} not found: {str(e)}", FailureLevel.ERROR)
```
Comment (enrico-usai) [inline cli/src/pcluster/validators/s3_validators.py]: Are you sure `bucket` variable is available in the except block?

Comment (hanwen-pcluste) [reply cli/src/pcluster/validators/s3_validators.py]: The variable is accessible

**Review by enrico-usai — DISMISSED** [review]

File: cli/src/pcluster/validators/s3_validators.py [hunk 2]
```diff
@@ -86,7 +86,14 @@ def _validate(self, url):
                 bucket = get_bucket_name_from_s3_url(url)
                 AWSApi.instance().s3.head_bucket(bucket_name=bucket)
             except AWSClientError as e:
-                self._add_failure(str(e), FailureLevel.ERROR)
+                if e.error_code == "403":
+                    self._add_failure(
+                        f"{str(e)}. Please attach a policy that allows the s3:ListBucket action for the resource "
+                        f"<ARN of bucket {bucket}> to the role or instance profile performing this operation",
```
Comment (charlesg3) [inline cli/src/pcluster/validators/s3_validators.py]: missing hard-stop `.` ?

**Review by charlesg3 — CHANGES_REQUESTED** [review]
Needs a period at the end of the message.

File: cli/src/pcluster/validators/s3_validators.py (hunk 2 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/validators/s3_validators.py]: Done

**Review by charlesg3 — APPROVED** [review]

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**